test-app/runtime/src/main/java/com/tns/Runtime.java

```diff
@@ -36,46 +36,69 @@
 import java.util.concurrent.RunnableFuture;
 import java.util.concurrent.atomic.AtomicInteger;
 
+import dalvik.annotation.optimization.CriticalNative;
+import dalvik.annotation.optimization.FastNative;
+
 public class Runtime {
+    @FastNative
     private native void initNativeScript(int runtimeId, String filesPath, String nativeLibDir, boolean verboseLoggingEnabled, boolean isDebuggable, String packageName,
                                          Object[] v8Options, String callingDir, int maxLogcatObjectSize, boolean forceLog);
 
+    @FastNative
     private native void runModule(int runtimeId, String filePath) throws NativeScriptException;
 
+    @FastNative
     private native void runWorker(int runtimeId, String filePath) throws NativeScriptException;
 
+    @FastNative
     private native Object runScript(int runtimeId, String filePath) throws NativeScriptException;
 
+    @FastNative
     private native Object callJSMethodNative(int runtimeId, int javaObjectID, String methodName, int retType, boolean isConstructor, Object... packagedArgs) throws NativeScriptException;
```

Comment:
not sure if these should be FastNative. FastNative completely disables GC while it's running, and these could likely run for a good while.

callJSMethodNative is even trickier because we can go java->native->java->native multiple times in long running methods

Reply:
So the fastest we can go for those would be `!`